mne/channels/channels.py

```diff
@@ -241,8 +241,9 @@ def _set_channel_positions(self, pos, names):
             msg = ('Channel positions must have the shape (n_points, 3) '
                    'not %s.' % (pos.shape,))
             raise ValueError(msg)
+        ch_names_set = set(self.ch_names)
```

Comment:
not sure this is more readable than what we did before. How does all this scale? Are our tests faster? I cannot really envision a scenario where this should otherwise matter if it completes on a microsecond or millisecond scale.